apiserver/facades/client/storage/register.go

```diff
@@ -16,32 +17,36 @@ import (
 // Register is called to expose a package of facades onto a given registry.
 func Register(registry facade.FacadeRegistry) {
 	registry.MustRegister("Storage", 6, func(stdCtx context.Context, ctx facade.ModelContext) (facade.Facade, error) {
-		return newStorageAPI(ctx) // modify Remove to support force and maxWait; add DetachStorage to support force and maxWait.
+		return newStorageAPI(stdCtx, ctx) // modify Remove to support force and maxWait; add DetachStorage to support force and maxWait.
 	}, reflect.TypeOf((*StorageAPI)(nil)))
 }
 
 // newStorageAPI returns a new storage API facade.
-func newStorageAPI(ctx facade.ModelContext) (*StorageAPI, error) {
+func newStorageAPI(stdCtx context.Context, ctx facade.ModelContext) (*StorageAPI, error) {
 	st := ctx.State()
-	model, err := st.Model()
-	if err != nil {
-		return nil, errors.Trace(err)
-	}
 
 	domainServices := ctx.DomainServices()
 	storageAccessor, err := getStorageAccessor(st)
 	if err != nil {
 		return nil, errors.Annotate(err, "getting backend")
 	}
 
+	modelType, err := domainServices.Model().ModelType(stdCtx, ctx.ModelUUID())
```

Comment:
You should ask this of the model info service where you can operate in the model context you are in.